llvm/lib/ProfileData/InstrProfReader.cpp

```diff
@@ -1303,6 +1303,10 @@ Error IndexedMemProfReader::deserializeV3(const unsigned char *Start,
   FrameBase = Ptr;
   CallStackBase = Start + CallStackPayloadOffset;
 
+  // Compute the number of elements in the radix tree array.
+  RadixTreeSize = (RecordPayloadOffset - CallStackPayloadOffset) /
```

Comment:
Sure.  I'll add a comment.  By the way, `RadixTreeSize` is used to reserve enough bits in a `BitVector`; I'm not computing the exact number of elements.  If I get a couple of extra elements because of padding, that's totally OK.  I'll include this in the comment also.

Reply:
Maybe add an assertion in the profile generator to note that the reader relies on this?

Reply:
What kind of assertion do you have in mind on the generator side?  Something like this?

```
// The deserializer relies on the fact that the record section immediately follows
// the radix tree array.
assert(CallStackPayloadOffset <= RecordPayloadOffset);
```

By the way, I've updated the comment in the latest revision.

Reply:
No, I'm concerned that the number RecordPayloadOffset has N elements but due to padding we compute that as N+1 elements when we compute the difference?

Reply:
The overestimation like that (that is, reality N elements, estimate N + 1 elements) is OK.  The only use of the estimate is to reserve enough bits in a BitVector introduced in this patch:

```
BitVector Worklist(RadixTreeSize);
```

I don't use the estimate to data structures allocated by other people.

Reply:
OK.  I've put an assert on the generator side in the latest revision.  Please take a look.  Thanks!